packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts

```diff
@@ -1,6 +1,65 @@
-import {Heading} from '@sanity/ui'
-import styled from 'styled-components'
+import {Box, Theme} from '@sanity/ui'
+import styled, {css} from 'styled-components'
 
-export const Title = styled(Heading)`
-  word-break: break-word;
+export const Root = styled(Box)`
+  position: relative;
 `
+
+export const Title = styled.h6(({theme}: {theme: Theme}) => {
+  const {fontSize, lineHeight} = theme.sanity.fonts.heading.sizes[5]
+
+  return css`
+    font-size: ${fontSize}px;
+    margin: 0;
+    line-height: ${lineHeight}px;
+    ${({
+      $titleHeight,
+      $isEditingForm,
+      $muted,
+    }: {
+      $titleHeight?: number
+      $isEditingForm?: boolean
+      $muted: boolean
+    }) => {
+      return css`
+        ${$titleHeight ? `height: ${$titleHeight}px` : ''};
+        ${$isEditingForm
+          ? css`
+              overflow: hidden;
+              text-overflow: ellipsis;
```

Comment:
`text-overflow: ellipsis` won't work in this case as it only applies to single-line text elements.
Also, it won't take effect if you allow the text to wrap, over when applying `overflow:hidden`.

I would suggest to try an implementation using CSS's 'line-clamp' or just Javascript.